pkg/agent/manager/cache/subscriber.go

```diff
@@ -33,10 +33,14 @@ type subscribers struct {
 }
 
 func NewSubscriber(selectors Selectors) (*subscriber, error) {
+	u, err := uuid.NewV4()
+	if err != nil {
+		return nil, err
```

Comment:
Heh... for your amusement, I present you with https://github.com/spiffe/spire/commit/7e3985875cb57e5813e271d036f5e1cc71658b18#diff-bc189db1d67c0b2efdefb255c26b6b3d

Reply:
heh ... who says acyclic graphs never have loops ;-)